### PR TITLE
Fix typos in mission dialogue

### PR DESCRIPTION
**Pull request by ThrawnCA**
-----------------------
**Bugfix:**

## Fix Details
Typo fixes

- Restore missing double quotes at end of dialogue, and remove some surplus ones.
- Fix some spelling and grammar errors.

## Testing Done
None.

### CODE REVIEW

**Comment by Amazinite** [comment]
You'll need to update the [codespell exclude file](https://github.com/endless-sky/endless-sky/blob/master/.codespell.exclude) for the one line that you changed that is not erroring on "vill."
Also, please refrain from eating the whitespace.

**Comment by ThrawnCA** [comment]
> Also, please refrain from eating the whitespace.

Sorry, I had Geany set to clean up whitespace by default. I've switched it off for this project now (and already had to go through reverting its attempts to replace tabs with spaces). I figured trimming a few trailing spaces was harmless, though. But that rule is now switched off for this project too.

File: data/hai/hai reveal 4 middle.txt [hunk 1]
```diff
@@ -372,7 +372,7 @@ mission "Hai Reveal: Pirate Troubles [1]"
 				`	"I'd be glad to help. What can I do?"`
 					goto explain
 				`	"What issue?"`
-			`	"I understand that you just defended this system from human pirates. That is the issue that needs resolved.`
+			`	"I understand that you just defended this system from human pirates. That is the issue that needs resolving.`
```
Comment (EjoThims) [inline data/hai/hai reveal 4 middle.txt:375]:
```suggestion
			`	"I understand that you just defended this system from human pirates. That is the issue that needs to be resolved.`
```

This isn't necessarily incorrect (especially as spoken dialogue), but this also isn't the simplest fix.

Suggesting the change to a more fitting fix, but dropping "to be" is such a very common convention in speech that I really think this one may be best left as is.

File: data/hai/hai reveal 5 philosophers.txt [hunk 2]
```diff
@@ -850,7 +850,7 @@ mission "Hai Reveal [B09] Scanning Devil-Hide"
 			label moon
 			action
 				set "Hai Reveal [B09] Scanning Devil-Hide: label moon"
-			`	The moon? Hmmm... okay, it's just an ordinary chunk of rock, but radiation from the planet is disturbing the sensors. I can only see a few dozen meters below ground, even though the scanner should be able to look down at least ten times that far... Aha!"`
+			`	"The moon? Hmmm... okay, it's just an ordinary chunk of rock, but radiation from the planet is disturbing the sensors. I can only see a few dozen meters below ground, even though the scanner should be able to look down at least ten times that far... Aha!`
```
Comment (EjoThims) [inline data/hai/hai reveal 5 philosophers.txt:853]: Like 846 above could use its trailing quotes clipped the same as this one, actually. :)

**Review by EjoThims — CHANGES_REQUESTED** [review]
Lots of good catches, just those couple of notes.

File: data/hai/hai reveal 4 middle.txt (hunk 1 above)
Comment (ThrawnCA) [reply data/hai/hai reveal 4 middle.txt:375]: I understand that "to be" is often dropped, but "needs resolved" is definitely not valid. "Needs to be resolved" would be fine.

Comment (EjoThims) [reply data/hai/hai reveal 4 middle.txt]: Spoken dialogue not being grammatically correct is wholly valid, as long as it is intended. ;)

**Review by EjoThims — APPROVED** [review]

**Review by MasterOfGrey — APPROVED** [review]
I’m quite impressed, and more than a little surprised, that you found that many. They all look good to me though.
Some of the white space removals in the middle of missions are unnecessary, but otherwise. Nicely done!

**Comment by ThrawnCA** [comment]
> I’m quite impressed, and more than a little surprised, that you found that many.

The missing double quote in https://github.com/endless-sky/endless-sky/pull/7940 inspired me to do a search for similar issues, so I came up with a regex:

    grep -r '^[^"]*"[^"]*$' data

to identify lines containing just one double quote mark.

And I noticed a handful of spelling errors while reviewing the results.

(Ideally it would search for any odd number of double quotes, though that would be more complex.)

> Nicely done!

Thanks :)

**Review by tibetiroka — APPROVED** [review]
The fixes are good, but I'm also not sure about the whitespace changes. I guess they can stay.

**Review by Zitchas — APPROVED** [review]
I haven't had the chance to review everything, but I have looked over the changes in the Remnant files, and they look good.

File: data/human/deep missions.txt [hunk 3]
```diff
@@ -1292,11 +1292,11 @@ mission "Deep: Remnant 0"
 		has "Terminus exploration: done"
 		has "First Contact: Hai: offered"
 		has "main plot completed"
-	
+
```
Comment (Amazinite) [inline data/human/deep missions.txt:1295]:
```suggestion
	
```

File: data/human/deep missions.txt [hunk 4]
```diff
@@ -2428,15 +2428,15 @@ mission "Deep: Scientist Rescue 3B"
...
 			`	When you look back to Lieutenant Paris, you notice that he is holding back tears. Laura Paris was one of the scientists who died.`
-			`	He turns to you and hands you a license, then says in a rather strained voice. "Thank you. I've said it before, but there's really no way I can thank you enough.`
+			`	He turns to you and hands you a license, then says in a rather strained voice. "Thank you. I've said it before, but there's really no way I can thank you enough."`
 			`	You look down at the license that Paris just handed you and notice that it's a city-ship license, one that will allow you to purchase the Bactrian. "I think it's only fair that you be able to purchase one after having to fight one to the death, and the government of the Deep agrees. It's really the least I... we could do for you."`
 			choice
 				`	"Thank you so much! I'll make sure that this gets put to good use."`
 				`	"Really? You shouldn't have."`
-			
+
```
Comment (Amazinite) [inline data/human/deep missions.txt:2436]:
```suggestion
			
```

File: data/remnant/remnant 1 introduction.txt [hunk 5]
```diff
@@ -1373,12 +1373,12 @@ mission "Remnant: Deep Surveillance"
 			`	Ivan opens the door with a big smile on his face. "Captain <last>! So great to see you here. Please, have a seat." Ivan ushers you inside his office, an impressively large room with a window wall overseeing a beautiful courtyard. You sit down in a cushioned chair on the opposite side of a desk. You lean back into the chair and get comfortable. "I know the chairs are cozy," Ivan says. "Don't fall asleep on me though."`
 			branch second
 				"retrieved information" == 2
-			`	Ivan then takes a seat in his own cushioned chair. "Welcome to my office. This is miles above what I'm used to. After gaining the attention of the Deep government with the discoveries we were making, they immediately had my office moved from the dingy complex it used to be at to here. I have you to thank for much of that work .`
+			`	Ivan then takes a seat in his own cushioned chair. "Welcome to my office. This is miles above what I'm used to. After gaining the attention of the Deep government with the discoveries we were making, they immediately had my office moved from the dingy complex it used to be at to here. I have you to thank for much of that work.`
 				goto reason
-			
+
```
Comment (Amazinite) [inline data/remnant/remnant 1 introduction.txt:1378]:
```suggestion
			
```

File: data/remnant/remnant 1 introduction.txt [hunk 6]
```diff
@@ -1373,12 +1373,12 @@ mission "Remnant: Deep Surveillance"
...
 			label second
 			`	Ivan then takes a seat in his own cushioned chair. "Welcome to my office. Much better than the old place, isn't it? After gaining the attention of the Deep government with the discoveries we were making, they immediately had my office moved from that dingy complex to here. I have you to thank for much of that work.`
-			
+
```
Comment (Amazinite) [inline data/remnant/remnant 1 introduction.txt:1381]:
```suggestion
			
```

File: data/sheragi/archaeology missions.txt [hunk 7]
```diff
@@ -648,16 +648,16 @@ mission "Sheragi Archaeology: The Box 2"
 	description "Take the mysterious box to a trusted scientist on <destination>."
 	source "Zug"
 	destination "Vinci"
-	
+
```
Comment (Amazinite) [inline data/sheragi/archaeology missions.txt:651]:
```suggestion
	
```

File: data/sheragi/archaeology missions.txt [hunk 8]
```diff
@@ -648,16 +648,16 @@ mission "Sheragi Archaeology: The Box 2"
...
 	to offer
 		has "Sheragi Archaeology: The Box 1: done"
-	
+
```
Comment (Amazinite) [inline data/sheragi/archaeology missions.txt:654]:
```suggestion
	
```

File: data/sheragi/archaeology missions.txt [hunk 9]
```diff
@@ -875,16 +875,16 @@ mission "Sheragi Archaeology: The Box 4"
 			label nomoney
 			`	"I know you don't. I can arrange for the payment to be made in different ways. Organ donation, prostitution, volunteering as a test subject... We'll need an appraisal first, though."`
 			`	You politely decline.`
-			`	"Come back to me when you have it, then, and I shall provide you with the information. We have all the time in the universe. Until you die, or someone decides to destroy the cube, that is.`
+			`	"Come back to me when you have it, then, and I shall provide you with the information. We have all the time in the universe. Until you die, or someone decides to destroy the cube, that is."`
 				goto exit
-			
+
```
Comment (Amazinite) [inline data/sheragi/archaeology missions.txt:880]:
```suggestion
			
```

File: data/sheragi/archaeology missions.txt [hunk 10]
```diff
@@ -875,16 +875,16 @@ mission "Sheragi Archaeology: The Box 4"
...
 			label time
 			`	"Come back when you're ready, then. We have all the time in the universe. Until you die, or someone decides to destroy the cube, that is. It's all up to you."`
-			
+
```
Comment (Amazinite) [inline data/sheragi/archaeology missions.txt:883]:
```suggestion
			
```

File: data/sheragi/archaeology missions.txt [hunk 11]
```diff
@@ -875,16 +875,16 @@ mission "Sheragi Archaeology: The Box 4"
...
 			label exit
 			`	They gesture, and the men promptly and diligently escort you back out to your ship.`
 				defer
-			
+
```
Comment (Amazinite) [inline data/sheragi/archaeology missions.txt:887]:
```suggestion
			
```

File: data/sheragi/archaeology missions.txt [hunk 12]
```diff
@@ -1417,27 +1417,27 @@ mission "Sheragi Archaeology: The Emerald Sword 3"
 			`	"Is this..." Eyama says, trailing off as she approaches one of the writings on the wall.`
 			choice
 				`	"What is it?"`
-			
+
```
Comment (Amazinite) [inline data/sheragi/archaeology missions.txt:1420]:
```suggestion
			
```

File: data/sheragi/archaeology missions.txt [hunk 13]
```diff
@@ -1417,27 +1417,27 @@ mission "Sheragi Archaeology: The Emerald Sword 3"
...
+			`	"From what we know, Rekla was a divine warrior among the Sheragi. Sheragi legends stated that she united the warring Sheragi tribes and led them all to the stars.`
 			label young
 			`	"This ship has to have traveled between the stars for many of your years in order to reach this system without a hyperdrive. The engines are powerful, so it perhaps took them 120 years to arrive here. Many of the Sheragi would have survived the trip, but would be very old when they arrived. This would have made it difficult for them to colonize the planet."`
 			choice
 				`	"So they would have needed to reproduce on the way here?"`
 				`	"Does that mean this was a daycare of sorts?"`
 					goto daycare
-			
+
```
Comment (Amazinite) [inline data/sheragi/archaeology missions.txt:1434]:
```suggestion
			
```

File: data/sheragi/archaeology missions.txt [hunk 14]
```diff
@@ -1417,27 +1417,27 @@ mission "Sheragi Archaeology: The Emerald Sword 3"
...
 			`	"Yes. At least one, maybe two generations of Sheragi could have been born in that time."`
 				goto walls
-				
+
```
Comment (Amazinite) [inline data/sheragi/archaeology missions.txt:1437]:
```suggestion
			
```

File: data/sheragi/archaeology missions.txt [hunk 15]
```diff
@@ -1417,27 +1417,27 @@ mission "Sheragi Archaeology: The Emerald Sword 3"
...
 			label daycare
 			`	Eyama walks over to one of the objects on the ground and picks it up. "These are toys. So yes. This does seem to have been converted into a daycare."`
-			
+
```
Comment (Amazinite) [inline data/sheragi/archaeology missions.txt:1440]:
```suggestion
			
```

**Pull request merged**